--- a/Source/JavaScriptCore/assembler/ARM64Registers.h
+++ b/Source/JavaScriptCore/assembler/ARM64Registers.h
@@ -43,7 +43,7 @@
 // beyond what you would save by the platform-agnostic "preserve return address" and "restore
 // return address" operations in CCallHelpers.
 
-#if OS(LINUX)
+#if OS(LINUX) && !defined(WTF_CRT_DARWIN_ARM64_ABI)
 #define FOR_EACH_GP_REGISTER(macro)                             \
     /* Parameter/result registers. */                           \
     macro(x0,  "x0"_s,  0, 0)                                     \
